### PR TITLE
#56 Added autocompletion for envDynawo.sh

### PR DESCRIPTION
**Pull request by gautierbureau**

### CODE REVIEW

File: util/envDynawo.sh [hunk 1]
```diff
@@ -598,6 +621,10 @@ build_tests() {
 }
 
 list_tests() {
+  if [ ! -d "$THIRD_PARTY_BUILD_DIR_VERSION" ]; then
```
Comment (rosiereflo) [inline util/envDynawo.sh]: Why is this required?

File: util/envDynawo.sh [hunk 2]
```diff
@@ -781,18 +808,37 @@ test_modelica_doc() {
...
+  fi
   $DYNAWO_INSTALL_DIR/bin/launcher --dydlib $*
   RETURN_CODE=$?
   return ${RETURN_CODE}
 }
 
-generate_preassembled_gdb() {
+dydLibGenerator_gdb() {
+  if [ "$BUILD_TYPE" != "Debug" ]; then
+    echo "You should compile with BUILD_TYPE=Debug."
```
Comment (rosiereflo) [inline util/envDynawo.sh]: We can use gdb in release mode. (and I do it on a regular basis)

File: util/envDynawo.sh [hunk 3]
```diff
@@ -1235,12 +1300,8 @@ case $MODE in
     build_3rd_party_version || error_exit "Error while building 3rd party version"
     ;;
 
-  compileModelicaOMCHelp)
```
Comment (rosiereflo) [inline util/envDynawo.sh]: Why is this removed?

File: util/envDynawo.sh [hunk 4]
```diff
@@ -71,17 +71,17 @@ where [option] can be:
     clean-build-3rd-party             clean then build 3rd party libraries
     version-validation                clean all built items, then build them all and run non-regression tests
     dump-model ([args])               call dumpModel executable with given arguments setting LD_LIBRARY_PATH correctly
-    compileModelicaOMC([args])        call compilerModelicaOMC with given arguments
-    generate-preassembled             generate a preassembled model
-    generate-preassembled-gdb         generate a preassembled model with debugger
+    compilerModelicaOMC([args])       call compilerModelicaOMC with given arguments
+    dydLibGenerator                   generate a preassembled model
```
[the remark this answers is not in the document]
Comment (gautierbureau) [reply util/envDynawo.sh]: I agree but in this case i would rename the bin to generate-preassembled because it is very confusing to have all those different names

File: util/envDynawo.sh [hunk 5]
```diff
@@ -465,16 +465,39 @@ build_3rd_party() {
...
 }
 
+# Very simple check to see if each third party has installed something or not
+is_3rd_party_version_installed() {
+  third_party_folders=(adept libiidm libxml libzip suitesparse sundials xerces-c)
+  for folder in ${third_party_folders[*]}; do
+    if [ ! -d "$THIRD_PARTY_INSTALL_DIR_VERSION/$folder" ]; then
+      return 1
+    fi
+    if [ ! -d "$THIRD_PARTY_INSTALL_DIR_VERSION/$folder/lib" ]; then
```
[the remark this answers is not in the document]
Comment (gautierbureau) [reply util/envDynawo.sh]: Ok to keep in mind for the future

File: util/envDynawo.sh [hunk 6]
```diff
@@ -522,14 +545,14 @@ config_dynawo() {
...
-# show the compiler Modelica help
-compileModelicaOMCHelp() {
-  $DYNAWO_INSTALL_DIR/bin/launcher --compile --help
-}
-
 # Compile a modelica model
-compileModelicaOMC() {
-  $DYNAWO_INSTALL_DIR/bin/launcher --compile $@ --remove-model-files false
+compiler_Modelica_OMC() {
+  if [ ! -x "$DYNAWO_INSTALL_DIR/bin/launcher" ]; then
```
[the remark this answers is not in the document]
Comment (gautierbureau) [reply util/envDynawo.sh]: Ok

File: util/envDynawo.sh (hunk 1 above)
Comment (gautierbureau) [reply util/envDynawo.sh]: Dynawo cmake config fails otherwise

File: util/envDynawo.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (rosiereflo) [reply util/envDynawo.sh]: The future is going to be checked-in today hopefully :)

File: util/envDynawo.sh (hunk 2 above)
Comment (gautierbureau) [reply util/envDynawo.sh]: ok

File: util/envDynawo.sh (hunk 3 above)
Comment (gautierbureau) [reply util/envDynawo.sh]: No need for this as compilerModelica --help does the same, I think that we have to much options and with the autocompletion --help will appear in the suggestions

**Comment by gautierbureau** [comment]
To maintain the scripts: 
- if it is a general option (like build-dynawo) with no option the completion will be automatically added if added in the usage of envDynawo. global_commands variable in the template_bash is based on the output of help command and then is updated if envDynawo changes.
- if the option (like dydLibGenerator) has options you have to add it mannualy. compilerModelicaOMC or dydLibGenerator arge good examples on how to add options. COMPREPLY is the most important variable and is the one that should be rightfully set to generate autocompletion.

File: util/envDynawo.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (gautierbureau) [reply util/envDynawo.sh]: Renaming will be done in #108
